tools/Google.Cloud.Tools.GenerateSBOM/Program.cs

```diff
@@ -1,329 +1,329 @@
-    {
-        /* convert UUID to local byte order */
-        var time_low = IPAddress.NetworkToHostOrder(
-            BitConverter.ToInt32(hash, 0));
-        var time_mid = IPAddress.NetworkToHostOrder(
-            BitConverter.ToInt16(hash, 4));
-        var time_hi_and_version = IPAddress.NetworkToHostOrder(
-            BitConverter.ToInt16(hash, 6));
-        var clock_seq_hi_and_reserved = hash[8];
-
-        /* put in the variant and version bits */
-        const short version = 5;
-        time_hi_and_version &= 0x0FFF;
-        time_hi_and_version |= version << 12;
-        clock_seq_hi_and_reserved &= 0x3F;
-        clock_seq_hi_and_reserved |= 0x80;
-        hash[8] = clock_seq_hi_and_reserved;
-
-        return new Guid(time_low, time_mid,
-            time_hi_and_version, hash[8..16]);
-    }
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text.Json;
+using System.Xml;
+
+namespace GenerateSBOM;
+
+// The classes below are not general purpose Spdx classes.  Their single
+// purpose is to generate json SBOMs for nupkgs generated from this repository,
+// so they have default values convenient for that purpose.
+// Reusing these classes for any other purpose would require a refactor.
+
+/// <summary>
+/// SPDX format.
+/// See https://github.com/spdx/spdx-spec/blob/development/v2.3.1/schemas/spdx-schema.json
+/// </summary>
+internal class Spdx
+{
+    public string spdxVersion { get; } = "SPDX-2.3";
+    public string dataLicense { get; } = "CC0-1.0";
+    public string SPDXID { get; } = "SPDXRef-Document";
+    public string name { get; set; } = "";
+
+    public string documentNamespace { get; set; } = "";
+    public List<string> creators { get; set; } = new List<string>();
+    public string created { get; set; } = "";
+
+    public List<SpdxPackage> packages { get; set; } = new List<SpdxPackage>();
+
+    public List<SpdxRelationship> relationships { get; set; }
+        = new List<SpdxRelationship> { new() };
+}
+
+internal class SpdxPackage
+{
+    public string name { get; set; } = "";
+    public string SPDXID { get; set; } = "";
+    public string versionInfo { get; set; } = "";
+    public string description { get; set; } = "";
+    public string supplier { get; set; } = "Person: Google LLC";
```

Comment:
Google isn't a person - I'd expect this to be Organization, according to https://spdx.github.io/spdx-spec/v2.3/package-information/#751-description

Reply:
Makes sense. I'm done for the day on my work laptop, but will send a message tomorrow.